apps/web/src/helpers/forms.ts

```diff
@@ -0,0 +1,9 @@
+import { inferServerActionInput, TAnyZodSafeFunctionHandler } from 'zsa'
+
+export function formDataToAction<
+  T extends (data: any) => Promise<[any, null] | [null, any]>,
+>(data: FormData): inferServerActionInput<TAnyZodSafeFunctionHandler & T> {
+  const obj: { [key: string]: unknown } = {}
+  data.forEach((value, key) => (obj[key] = value))
+  return obj as inferServerActionInput<TAnyZodSafeFunctionHandler & T>
+}
```

Comment:
We have like 3 abstractions for actions. I would use `useLatitudeAction` in your code instead of doing one more different way of working with actions.

Reply:
But I need to get the data from the form before executing the actual action somehow...! Otherwise, the default behaviour of the uncontrolled model input, of the create provider modal, will send the model as an empty string when the input is empty.

Reply:
I was using `useFormAction`, and modified it to be able to change the captured payload, but I was told not to modify that hook and make the code in the component. However, with the`formDataToJson` function of the `useFormAction` I was not getting any types, so I decided to move that function outside of the `useFormAction` file and enhance it to allow for type safety/inference....

Reply:
Just to clarify this function is just to transform the form data to a JS object. If I am not mistaken, even when using useLatitudeAction, I will still have to use it.... I use this function before calling the action:

```typescript
const { create } = useProviderApiKeys()
  const onSubmit = useCallback(
    async (event: FormEvent<HTMLFormElement>) => {
      event.preventDefault()

      const payload = formDataToAction<typeof create>(
        new FormData(event.currentTarget),
      )

      const [_, error] = await create({
        ...payload,
        defaultModel: payload.defaultModel || undefined, // Ensure defaultModel is either non-empty or undefined
      })
      if (!error) onOpenChange(false)
    },
    [create],
  )
  ```

Reply:
this is good 👌🏼